enclave/core/CMakeLists.txt

```diff
@@ -138,7 +138,9 @@ target_compile_options(oecore PUBLIC
 # Enclaves are linked using -pie and therefore global-dynamic is too conservative.
 # Of the two efficient static tls models, inital-exec and local-exec, we choose the most
 # optimal one.
-    -ftls-model=local-exec)
+    -ftls-model=local-exec
+    # Disable builtin functions for enclaves, but only in our build.
+    $<BUILD_INTERFACE:-fno-builtin-malloc -fno-builtin-calloc -fno-builtin>)
```

Comment:
Does $<BUILD_INTERFACE:flags> mean that the flags will be applied when building the module as well, in addition to being applied to dependents of the module?

Reply:
Yes; it means "apply as normal when building our code, but don't include in an `install` command." That is, it excludes it from the CMake package, but that's it.

> $<BUILD_INTERFACE:...>
Content of ... when the property is exported using export(), or **when the target is used by another target in the same buildsystem.** Expands to the empty string otherwise.

Reply:
Ok.
While we are in the area, It would be good to capture the rationale for why we want to disable "builtin functions for enclaves in our build; but not for user code".  It is not readily clear why we would want to do it.

Also that would mean that we will be compiling all our enclaves with a set of flags; but expect the user to use another set of flags which we don't lock down.

Reply:
That's a good question that I don't have a great answer for, I was just resolving an outstanding TODO. I believe you had requested it to be only in our CMake package; can you provide an explanation?

Reply:
@johnkord Can you provide the rationale for these options?